CHANGELOG.md

```diff
@@ -1,11 +1,54 @@
+
+In this minor version release, support for Python 3.9 is added [[#4301]](https://github.com/aiidateam/aiida-core/pull/4301), while support for Python 3.5 is dropped [[#4386]](https://github.com/aiidateam/aiida-core/pull/4386).
+This version is compatible with all current Python versions that are not end-of-life:
+ * 3.6
+ * 3.7
+ * 3.8
+ * 3.9
+
+### Features
+- Process functions (`calcfunction` and `workfunction`) can now be submitted to the daemon just like `CalcJob`s and `WorkChain`s [[#4539]](https://github.com/aiidateam/aiida-core/pull/4539)
+- REST API: list endpoints at base URL [[#4412]](https://github.com/aiidateam/aiida-core/pull/4412)
+- `ProcessBuilder`: allow unsetting of inputs through attribute deletion [[#4419]](https://github.com/aiidateam/aiida-core/pull/4419)
+- `verdi migrate`: make `--in-place` work across different file systems [[#4393]](https://github.com/aiidateam/aiida-core/pull/4393)
+
+### Improvements
+- Added remaining original documentation that didn't make it into the first step of the recent major overhaul of v1.3.0
+- `verdi process show`: order by ctime and print process label [[#4407]](https://github.com/aiidateam/aiida-core/pull/4407)
+- `LinkManager`: fix inaccuracy in exception message for non-existent link [[#4388]](https://github.com/aiidateam/aiida-core/pull/4388)
+- Add `reset` method to`ProgressReporterAbstract` [[#4522]](https://github.com/aiidateam/aiida-core/pull/4522)
+- Improve the deprecation warning for `Node.open` outside context manager [[#4434]](https://github.com/aiidateam/aiida-core/pull/4434)
+
+### Bug fixes
+- Fix `ZeroDivisionError` in worker slots check [[#4513]](https://github.com/aiidateam/aiida-core/pull/4513)
+- `CalcJob`: only attempt to clean up the retrieve temporary folder after parsing if it is present [[#4379]](https://github.com/aiidateam/aiida-core/pull/4379)
+- Add missing entry point groups to the mapping [[#4395]](https://github.com/aiidateam/aiida-core/pull/4395)
+
+### Developers
+- Replace old format string interpolation with f-strings [[#4400]](https://github.com/aiidateam/aiida-core/pull/4400)
+- CI: move `pylint` configuration to `pyproject.toml` [[#4411]](https://github.com/aiidateam/aiida-core/pull/4411)
+- CI: use `-e` install for tox + add docker-compose for isolated RabbitMQ [[#4375]](https://github.com/aiidateam/aiida-core/pull/4375)
+- CI: add coverage patch threshold to prevent false positives [[#4413]](https://github.com/aiidateam/aiida-core/pull/4413)
+- Archive importer: refactor the reading code [[#4510]](https://github.com/aiidateam/aiida-core/pull/4510)
+- Archive importer: refactor the migrations code [[#4532]](https://github.com/aiidateam/aiida-core/pull/4532)
+- Archive exporter: refactor progress bar [[#4504]](https://github.com/aiidateam/aiida-core/pull/4504)
+- Archive exporter: decouple the extraction of data from the writing to file [[#4448]](https://github.com/aiidateam/aiida-core/pull/4448)
```

Comment:
```suggestion
- Archive (import/export) refactor:
     * The refactoring goal was to pave the way for the implementation of a new archive format in v2.0.0 ([ aiidateam/AEP005](https://github.com/aiidateam/AEP/pull/21))
     * Three abstract+concrete interface classes are defined; writer, reader, migrator, which are **independent of the internal structure of the archive**. These classes are used within the export/import code.
     * The code in `aiida/tools/importexport` has been largely re-written, in particular adding `aiida/tools/importexport/archive`, which contains this code for interfacing with an archive, and **does not require connection to an AiiDA profile**.
     * The export logic has been re-written; to minimise required queries (faster), and to allow for "streaming" data into the writer (minimise RAM requirement with new format). It is intended that a similiar PR will be made for the import code.
     * A general progress bar implementation is now available in `aiida/common/progress_reporter.py`. All corresponding CLI commands now also have `--verbosity` option.
     * Merged PRs:
         * Refactor export archive ([#4448](https://github.com/aiidateam/aiida-core/pull/4448) & [#4534](https://github.com/aiidateam/aiida-core/pull/4534))
         * Refactor import archive ([#4510](https://github.com/aiidateam/aiida-core/pull/4510))
         * Refactor migrate archive ([#4532](https://github.com/aiidateam/aiida-core/pull/4532))
         * Add group extras to archive ([#4521](https://github.com/aiidateam/aiida-core/pull/4521))
         * Refactor cmdline progress bar ([#4504](https://github.com/aiidateam/aiida-core/pull/4504) & [#4522](https://github.com/aiidateam/aiida-core/pull/4522))
       * Deprecations: `export_zip`, `export_tar`, `export_tree`, `extract_zip`, `extract_tar` and `extract_tree` functions. `silent` key-word in the `export` function
       * Removed: `ZipFolder` class
```

Reply:
Since it is a pretty big change with a considerable explanation, I think it would be better to make it a separate section. Check out how it looks now and let me know what you think.

Reply:
yep thats fine ta 👍